auth.go

```diff
@@ -207,6 +209,10 @@ func authenticate(
 		requestMain.Authenticator = externalBrowserAuthenticator
 	} else if !bytes.Equal(samlResponse, []byte{}) {
 		requestMain.RawSAMLResponse = string(samlResponse)
+	} else if token != "" {
```

Comment:
This checking may not be good enough?

Reply:
I think it's ok for now since token is a string and you can't have nil for a string:

```
var s string
fmt.Println(s=="") // prints "true"
```

Provided that the token is at least one character, we'll go into that else if.

Reply:
I mean token != "" doesn't necessarily oAuth authenticator, right?

Reply:
I think right now a token that != "" means that the user is attempting to use OAuth since the golang driver does not support any other token based authentication mechanisms. What makes the `authenticate` call in `auth.go` a little unclean here is that "authenticator" is never passed in to authenticate - we just take in every possible value that each authenticator uses and infer which authenticator to use based on the values. For instance, `samlResponse` not being an empty byte array  implies Okta, `proofKey` not being an empty byte array implies external browser authentication, etc.

Reply:
ok, fair enough. I agree that we need to clean up those checkings.

Reply:
I can move some stuff around... let me create another PR with those changes!